pkg/check/check_iam.go

```diff
@@ -0,0 +1,126 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package check
+
+import (
+	"context"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var (
+	DeployNames = []string{"ibm-iam-operator", "auth-idp", "auth-pap", "auth-pdp", "configmap-watcher", "iam-policy-controller", "oidcclient-watcher", "secret-watcher"}
+	JobNames    = []string{"iam-onboarding", "security-onboarding", "oidc-client-registration"}
+)
+
+func IamStatus(mgr manager.Manager) {
+	reader := mgr.GetAPIReader()
+	client := mgr.GetClient()
+
+	for {
+		iamStatus := overallIamStatus(reader)
+		if err := createUpdateConfigmap(reader, client, iamStatus); err != nil {
+			klog.Error("create or update configmap failed")
+		}
+		time.Sleep(2 * time.Minute)
+	}
+}
+
+func overallIamStatus(reader client.Reader) string {
+	for _, deploy := range DeployNames {
+		status := getDeploymentStatus(reader, deploy)
+		if status == "NotReady" {
+			return status
+		}
+	}
+	for _, job := range JobNames {
+		status := getJobStatus(reader, job)
+		if status == "NotReady" {
+			return status
+		}
+	}
+	return "Ready"
+}
+
+func getJobStatus(client client.Reader, name string) string {
+	job := &batchv1.Job{}
+	jobName := name
+	jobNs := "ibm-common-services"
+	err := client.Get(context.TODO(), types.NamespacedName{Name: jobName, Namespace: jobNs}, job)
+	if err != nil {
+		return "NotReady"
+	}
+
+	if job.Status.Succeeded >= *job.Spec.Completions {
+		return "Ready"
+	}
+	return "NotReady"
+}
+
+func getDeploymentStatus(reader client.Reader, name string) string {
+	deploy := &appsv1.Deployment{}
+	deployName := name
+	deployNs := "ibm-common-services"
+
+	err := reader.Get(context.TODO(), types.NamespacedName{Name: deployName, Namespace: deployNs}, deploy)
+	if err != nil {
+		return "NotReady"
+	}
+
+	if deploy.Status.ReadyReplicas != deploy.Status.Replicas {
+		return "NotReady"
+	}
+	return "Ready"
+}
+
+func createUpdateConfigmap(reader client.Reader, client client.Client, status string) error {
+	cm := &corev1.ConfigMap{}
+	cmName := "ibm-common-services-status"
+	cmNs := "kube-public"
+	if status == "NotReady" {
+		klog.Info("IAM status is NoReady, waiting some minutes...")
+	}
+	err := reader.Get(context.TODO(), types.NamespacedName{Name: cmName, Namespace: cmNs}, cm)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			cm.Name = cmName
+			cm.Namespace = cmNs
+			cm.Data = make(map[string]string)
+			cm.Data["iamstatus"] = status
+			if err := client.Create(context.TODO(), cm); err != nil {
+				return err
+			}
+		}
+		return err
+	}
+	if cm.Data["iamstatus"] != status {
```

Comment:
Let's only monitor the bootstrap status, this operator can't monitor the runtime status.

So please remove this update function.